onnxruntime/core/providers/webgpu/quantization/quantize_linear.cc

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <vector>
+
+#include "core/util/math.h"
+#include "core/providers/webgpu/quantization/quantize_linear.h"
+#include "core/providers/webgpu/shader_helper.h"
+#include "core/providers/webgpu/webgpu_supported_types.h"
+#include "core/providers/webgpu/webgpu_utils.h"
+
+namespace onnxruntime {
+namespace webgpu {
+
+Status DequantizeLinearProgram::GenerateShaderCode(ShaderHelper& shader) const {
+  const auto& x = shader.AddInput("input", ShaderUsage::UseUniform | ShaderUsage::UseIndicesTypeAlias | ShaderUsage::UseElementTypeAlias);
+  const auto& scale = shader.AddInput("scale", ShaderUsage::UseUniform | ShaderUsage::UseIndicesTypeAlias | ShaderUsage::UseValueTypeAlias);
+  const auto& output = shader.AddOutput("output", ShaderUsage::UseUniform | ShaderUsage::UseShapeAndStride | ShaderUsage::UseValueTypeAlias);
+
+  shader.MainFunctionBody()
+      << shader.GuardAgainstOutOfBoundsWorkgroupSizes("uniforms.output_size")
+      << "let output_indices = " << output.OffsetToIndices("global_idx") << ";\n";
+
+  // Get x input
+  if (packed_) {
+    std::string unpack = (signed_) ? "unpack4xI8(x)" : "unpack4xU8(x)";
+    if (output.NumComponents() == 1) {
+      shader.MainFunctionBody()
```

Comment:
Unclear why we are dividing by 4 just by reading the code, can you add a comment?

Reply:
can add a comment.

Reply:
let me merge this one - will add the comment in the next PR